specs/transport/car/carv2/index.md

```diff
@@ -0,0 +1,177 @@
+
+## Format Description
+
+CARv2 consists of:
+
+1. An 11-byte pragma (or sequence of "magic bytes") that identify the data as a CARv2 format.
+2. A header describing data payload offset and length and index offset within the container as well as a bitfield for setting various capabilities supported by the container.
+3. A standard CARv1 data payload, including standard CARv1 header and roots and sequence of `CID:Bytes` pairs.
+4. An optional index payload, which may be one of a number of supported index formats, allowing for fast lookups of blocks within the data payload in the form of an offset from the start of the container.
+
+The CARv2 format can be illustrated as follows:
+
+```
+| 11-byte fixed pragma | 40-byte header | optional padding | CARv1 data payload | optional padding | optional index payload
+```
+
+### Pragma
+
+The CARv2 version pragma (or "magic bytes") was chosen for compatibility with existing CARv1 parsers. CARv1 leads with a [DAG-CBOR](../../../codecs/dag-cbor/index.md) block, prefixed with a varint where the block contains the version and roots array:
+
+```ipldsch
+type CarV1Header struct {
+  version Int # 1
+  roots [&Any]
+}
+```
+
+To introduce a new version that existing parsers can safely reject as an *"unsupported version"* we must copy the minimal form of this header with a new version.
+
+Therefore, the CARv2 header is a fixed sequence of 11 bytes: **`0x0aa16776657273696f6e02`**.
+
+These bytes decode as follows: A leading `0x0a` which translates as a `uint(10)` (or `varint(10)`) indicating the length of the DAG-CBOR header block to follow. The remaining 10 bytes are a standard CBOR encoding of a map containing a `"version"` field with a value `2`. i.e.
+
+
+```ipldsch
+type CarV2Header struct {
+  version Int # 2
+}
+```
+
+The 10 bytes are parsed in CBOR as follows:
+
+```
+a1                                                # map(1)
+  67                                              #   string(7)
+    76657273696f6e                                #     "version"
+  02                                              #   uint(2)
+```
+
+Existing CARv1 parsers should safely read this pragma and reject the container as an unsupported version of the CAR format.
+
+This 11 byte string remains fixed and may be matched using a simple byte comparison and does not require a varint or CBOR decode since it does not vary for CARv2 containers.
+
+### Header
+
+Following the 11 byte pragma, the CARv2 is a fixed-length sequence of 40 bytes, broken into the following sections:
+
+1. **Characteristics**: A 128-bit (16-byte) bitfield used to describe certain features of the enclosed data.
```

Comment:
Do we want to clarify that this will begin as all zeros, given that we're not using it yet?

This is important because, if we add any characteristic bits in the future, we have to keep in mind that all past CARv2 blobs will have the bits unset, for backwards compatibility.

Reply:
:thumbsup:

> All bits in the bitfield will be unset (`0`) by default and only set (`1`) where they are being used to signal a characteristic other than the default.